### DISCUSSION TITLE
Upgrade to Spring Boot 2.4?

### DISCUSSION THREAD

**Discussion by donbeave** [graphql-java-kickstart/graphql-spring-boot#511] [Ideas]
Hello!

Do you have a plan to upgrade this library to Spring Boot 2.4?

We tried to upgrade some of our apps to SB 2.4.0, but found out it's not working because there are 2 dependant libraries that are not upgraded yet, one of them is [grpc-spring-boot-starter](https://github.com/yidongnan/grpc-spring-boot-starter/issues/469) (which is planned to upgrade) and the second one is this graphql-spring-boot-starter.

**Comment by oliemansm** [comment]
Good one. We're about to release a new version supporting graphql-java 16.1 too. We'll add the Spring Boot upgrade in that one.

**Comment by donbeave** [comment]
Will you update the Kotlin version as well?

**Reply by oliemansm** [reply]
Think we could unless there are some major breaking changes making it not that easy to upgrade.

**Reply by donbeave** [reply]
I'm not sure about breaking changes. But the latest Spring Boot 2.4.1 already upgraded to Kotlin 1.4: https://github.com/spring-projects/spring-boot/issues/24391.

Probably this also needs to be resolved first: https://github.com/graphql-java-kickstart/graphql-java-tools/issues/451. As I remember when we did upgrade to Spring Boot 2.4.0, `graphql-spring-boot-starter` didn't work well because of the lower Kotlin version in `graphql-java-tools`.

**Reply by oliemansm** [reply]
Then I'm worried we might not even be able to upgrade Spring Boot to 2.4 yet, since that might cause some trouble. If you're willing and have time we're open to PRs on both projects.

**Reply by donbeave** [reply]
Yeah, agree. Let's do it. I think I can help with an upgrade if needed.

**Reply by oliemansm** [reply]
Seems not too much trouble:
* PR for tools: https://github.com/graphql-java-kickstart/graphql-java-tools/pull/473
* 11.1.0-SNAPSHOT of this library already upgraded.

**Reply by donbeave** [reply]
That's cool!!!
